kernel/driver.cc

```diff
@@ -244,11 +244,15 @@ int main(int argc, char **argv)
 	bool mode_q = false;
 
 #if defined(YOSYS_ENABLE_READLINE) || defined(YOSYS_ENABLE_EDITLINE)
-	if (getenv("HOME") != NULL) {
-		yosys_history_file = stringf("%s/.yosys_history", getenv("HOME"));
-		read_history(yosys_history_file.c_str());
-		yosys_history_offset = where_history();
+	if (getenv("XDG_STATE_HOME") == NULL || getenv("XDG_STATE_HOME")[0] == '\0') {
+		if (getenv("HOME") != NULL) {
+			yosys_history_file = stringf("%s/.local/state/.yosys_history", getenv("HOME"));
+		}
+	} else {
+		yosys_history_file = stringf("%s/.yosys_history", getenv("XDG_STATE_HOME"));
 	}
+	read_history(yosys_history_file.c_str());
```

Comment:
I think this will fail if neither `HOME` nor `XDG_STATE_HOME` are set. In that case you should not read the history file at all.